src/pages/instances/forms/UploadExternalFormatFileForm.tsx

```diff
@@ -0,0 +1,333 @@
+            `/ui/project/${project?.name}/instance/${instanceName}/configuration`,
+          ),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const handleFailure = (msg: string) => {
+    toastNotify.failure("Instance creation failed.", new Error(msg));
+  };
+
+  const invalidateCache = () => {
+    void queryClient.invalidateQueries({
+      predicate: (query) => {
+        return query.queryKey[0] === queryKeys.instances;
+      },
+    });
+  };
+
+  const handleConfirmUpload = (values: UploadExternalFormatFileFormValues) => {
+    if (!values.imageFile) {
+      return;
+    }
+
+    // start create instance operation
+    createInstance(uploadExternalFormatFilePayload(values), project?.name || "")
+      .then((operation) => {
+        const operationId = operation.metadata.id;
+        const operationSecret = operation.metadata?.metadata?.fs;
+        const instanceName = getInstanceName(operation.metadata);
+
+        // establish websocket connection based on the instance creation operation
+        let wsUrl = `wss://${location.host}/1.0/operations/${operationId}/websocket`;
+        if (operationSecret) {
+          wsUrl += `?secret=${operationSecret}`;
+        }
+
+        const ws = new WebSocket(wsUrl);
+
+        ws.onopen = () => {
+          sendFileByWebSocket(
+            ws,
+            formik.values.imageFile,
+            FILE_CHUNK_SIZE,
+            handleUploadProgress.bind(null, instanceName),
+            handleUploadError,
+          );
+        };
+
+        // set up event queue for the operation
+        eventQueue.set(
+          operationId,
+          () => handleSuccess(getInstanceName(operation.metadata)),
+          handleFailure,
+          invalidateCache,
+        );
+      })
+      .catch((e) => {
+        close();
+        toastNotify.failure("Instance creation failed.", e);
+      });
+  };
+
+  const formik = useFormik<UploadExternalFormatFileFormValues>({
+    initialValues: {
+      name: "",
+      pool: "",
+      imageFile: null,
+      formatConversion: true,
+      virtioConversion: false,
+      architecture: supportedVMArchOptions(
+        settings?.environment?.architectures || [],
+      )[0].value,
+    },
+    validateOnMount: true,
+    validationSchema: Yup.object().shape({
+      name: instanceNameValidation(
+        project?.name || "",
+        instanceNameAbort,
+      ).optional(),
+    }),
+    onSubmit: handleConfirmUpload,
+  });
+
+  const handleFileChange = async (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = formik.getFieldProps("imageFile");
+    onChange(e);
+
+    if (e.currentTarget.files) {
+      const file = e.currentTarget.files[0];
+      const suffix = "-import";
+      const fileExtension = getFileExtension(file.name);
+      const sanitisedFileName = sanitizeInstanceName(
+        file.name.split(fileExtension)[0],
+      );
+      const instanceName = truncateInstanceName(sanitisedFileName, suffix);
```

Comment:
When I select a file without a file extension, named "foo", I get "f-import", would expect "foo-import" for that case.

I think we don't need the file extension, maybe simplify this to be 1 function, that receives a filename and returns an instance name?

The other upload type uses "imported" suffix and uses a slightly different way to calculate the name. With a generic function, we could reuse it over there.